src/App.jsx

```diff
@@ -14,45 +16,124 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_BY_ALPHABET = 'alphabet';
+const SORT_BY_LENGTH = 'length';
+
+function getPrepearedGoods(goods, sortField, sortRevers) {
+  const prepearedGoods = [...goods];
+
+  if (sortField === SORT_BY_ALPHABET) {
+    prepearedGoods.sort((a, b) => {
+      switch (sortRevers) {
+        case true:
+          return b.localeCompare(a);
+        case false:
+          return a.localeCompare(b);
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (sortField === SORT_BY_LENGTH) {
+    prepearedGoods.sort((a, b) => {
+      switch (sortRevers) {
+        case true:
+          return b.length - a.length;
+        case false:
+          return a.length - b.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (sortField === '' && sortRevers) {
+    prepearedGoods.reverse();
+  }
+
+  return prepearedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [sortRevers, setSortRevers] = useState(false);
+
+  const visibleGoods = getPrepearedGoods(
+    goodsFromServer, sortField, sortRevers,
+  );
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const reverse = () => (
+    sortRevers ? (
+      setSortRevers(false)
+    ) : (
+      setSortRevers(true)
+    )
+  );
+
+  const reset = () => {
+    if (sortField !== '' || sortRevers) {
+      return true;
+    }
+
+    return false;
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={
+            cn(
+              'button is-info',
+              { 'is-light': sortField !== SORT_BY_ALPHABET },
+            )}
+          onClick={() => setSortField(SORT_BY_ALPHABET)}
```

Comment:
better to make one handler for two sortField cases which will accept sorField and return handler
```
  const makeSetSortField = (field) => () => setSortField(field);
```
```suggestion
          onClick={makeSetSortField(SORT_BY_ALPHABET)}
```